.github/workflows/main-build.yml

```diff
@@ -110,6 +117,159 @@ jobs:
           COMMIT_HASH: ${{ github.sha }}
       - uses: codecov/codecov-action@v3
 
+      - name: Get ADOT_JAVA_AGENT Image Tag
+        id: java_agent_versioning
+        working-directory: aws-otel-java-instrumentation
+        run: |
+          shortsha="$(git rev-parse --short HEAD)"
+          java_agent_tag=${{ env.ADOT_JAVA_VERSION }}-$shortsha
+          echo "STAGING_TAG=java_agent_tag" >> $GITHUB_OUTPUT
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2.1.0
+        with:
+          role-to-assume: ${{ secrets.JAVA_INSTRUMENTATION_SNAPSHOT_ECR }}
+          aws-region: us-west-2
+
+      - name: Login to private staging ecr
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.STAGING_ECR }}
+        env:
+          AWS_REGION: us-west-2
+
+      - name: Build and push staging image for e2e integration test
```

Comment:
The image generated using `otelagent` `jib` doesn't have `rustcp` utility and hence not useable for Operator testing. We do need to publish the image for java agent (with rust cp utility) from somewhere so that it can be pulled by the e2e testcase. This image of `staging` is built from `Dockerfile` and is first tested with `test-tag` before being published for `staging` for further e2e tests. Since we have a `test-tag` image step in the same job, so the Docker doesn't  rebuild the same image from scratch again but pull it out from local cache of `test-tag`. 
The step `Build and push staging image for e2e integration test` is publishing images for sample_app testing with java agent. I have now separated with step as it was not related to rest and renamed it accordingly.